sdk/documenttranslation/azure-ai-documenttranslation/samples/async_samples/sample_batch_translation_with_storage_async.py

```diff
@@ -0,0 +1,141 @@
+            source_storage_endpoint,
+            container_name=source_storage_container_name,
+            credential=source_storage_key
+        )
+
+        # upload some document for translation
+        with open("document.txt", "rb") as doc:
+            container_client.upload_blob("document.txt", doc)
+
+        # prepare translation job input
+        source_container_sas = generate_container_sas(
+            account_name=source_storage_account_name,
+            container_name=source_storage_container_name,
+            account_key=source_storage_key,
+            permission=ContainerSasPermissions.from_string("rl")
+        )
+
+        target_container_sas = generate_container_sas(
+            account_name=target_storage_account_name,
+            container_name=target_storage_container_name,
+            account_key=target_storage_key,
+            permission=ContainerSasPermissions.from_string("rlwd")
+        )
+
+        source_container_url = source_storage_endpoint + "/" + source_storage_container_name + "?" + source_container_sas
+        target_container_url = target_storage_endpoint + "/" + target_storage_container_name + "?" + target_container_sas
+
+        batch = [
+            BatchDocumentInput(
+                source_url=source_container_url,
+                source_language="en",
+                targets=[
+                    StorageTarget(
+                        target_url=target_container_url,
+                        language="es"
+                    )
+                ],
+                prefix="document"
+            )
+        ]
+
+        # run job
+        async with translation_client:
+            job_detail = await translation_client.create_translation_job(batch)
+            job_result = await translation_client.wait_until_done(job_detail.id)
+
+            # poll status result
+            if job_result.status == "Succeeded":
+                print("We translated our documents!")
+                if job_result.documents_failed_count > 0:
+                    check_documents(translation_client, job_result.id)
+
+            if job_result.status in ["Failed", "ValidationFailed"]:
+                if job_result.error:
+                    print("Translation job failed: {}: {}".format(job_result.error.code, job_result.error.message))
+                check_documents(translation_client, job_result.id)
+                exit(1)
+
+            # store result documents
+            container_client = ContainerClient(
+                target_storage_endpoint,
+                container_name=target_storage_container_name,
+                credential=target_storage_key
+            )
+
+            target_container_client = container_client.from_container_url(target_container_url)
+
+            with open("translated.txt", "wb") as my_blob:
+                download_stream = target_container_client.download_blob("document.txt")
+                my_blob.write(download_stream.readall())
```

Comment:
This will look a little different here for async:
```
download_stream = target_container_client.download_blob("document.txt")
data = await download_stream.readall()
my_blob.write(data)
```